packages/evm/src/evmAssetTransfer.ts

```diff
@@ -0,0 +1,107 @@
+import type { Config } from '@buildwithsygma/core';
+import { isValidAddressForNetwork } from '@buildwithsygma/core';
+import { Bridge__factory } from '@buildwithsygma/sygma-contracts';
+import { Web3Provider } from '@ethersproject/providers';
+import { constants, utils } from 'ethers';
+
+import { EvmTransfer } from './evmTransfer.js';
+import type { EvmAssetTransferParams, EvmFee, TransactionRequest } from './types.js';
+import { assetTransfer } from './utils/index.js';
+import { createTransactionRequest } from './utils/transaction.js';
+
+/**
+ * Asset transfers in EVM
+ * are of supported standards
+ * like ERC20, ERC721 and ERC1155
+ * which require set of approval and
+ * transfer transactions
+ * TODO: Add Support for all
+ */
+interface IAssetTransfer {
+  getTransferTransaction(): Promise<TransactionRequest>;
+  getApprovalTransactions(): Promise<Array<TransactionRequest>>;
+}
+
+/**
+ *
+ */
+export abstract class AssetTransfer extends EvmTransfer implements IAssetTransfer {
+  // amount in case if its a fungible transfer
+  protected specifiedAmount?: bigint;
```

Comment:
could it be optional?

Reply:
all of these are concretely declared in subclass